Fixes pyprobml-review/pyprobml-review#18

nitish1295 opened the pull request:
## Description

Added gauss_infer_1d.ipynb for [gauss_infer_1d.ipynb](https://github.com/probml/pyprobml/blob/master/notebooks/book1/03/gauss_infer_1d.ipynb)

<!-- Please refer to https://github.com/probml/pyprobml/blob/master/CONTRIBUTING.md before opening this PR -->

### Figure Number
Book 1 - 3.8

<!-- Mention figure number with corresponding book (book1 or book2). For example: book1 - 4.18 -->

### Figures
<!-- Create a public gist with rendered figures and add a link below. 
Refer to this template for example: https://gist.github.com/patel-zeel/28a101009f5e938082b6d93101bf1fd7 -->

Kindly note, that for now, I have added the figures here since it is easier to keep track and review. Will create a gist for master repo

| Before PR |![image](https://user-images.githubusercontent.com/21181046/165751183-6f0f9828-f16d-45db-8488-6c237e73154e.png) |
| :-: | :-: |
| After PR | ![image](https://user-images.githubusercontent.com/21181046/165751247-3f7a2fa6-145d-4507-871d-1a728278ec7f.png) |


### Issue
- Pyprobml Issue https://github.com/probml/pyprobml/issues/739

<!-- Link the issue you are solving -->

<!-- If the issue is from this repo, include directly with issue number -->
<!-- For example: 

#12

-->

<!-- If the issue is from another repo, you can include it using the regular linking mechanism -->
<!-- For example:

[Issue title](Issue link) 

-->

## Steps

<!-- If the task is complicated, you can specify each step using checkboxes so that everyone can understand what you have done and what you are going to complete. -->
<!-- For example:
-->

- [x] Changed the name of the variables/functions to PEP 8 style
- [x] Added code to generate two figures as given in the book and Latexified them.
- [x] Converted code to JAX
- [x] Added interactive demo to change variance, mean

## Checklist:

- [x] Performed a self-review of the code
- [x] Tested on Google Colab.

## Potential problems/Important remarks

<!-- If you have any important remarks for the reviewers to take a closer look, you can write them in detail here.  -->

- I have kept the y-label despite the axis being same since I feel compared to last time for https://github.com/pyprobml-review/pyprobml-review/pull/15 this y-axis label is slightly more meaningful, similar to the first plot in https://github.com/pyprobml-review/pyprobml-review/pull/8
-  For now, I have kept the captions the same as in the book, will try to update them in case I come across something more meaningful. Please let me know if you have any suggestions
- Updated x-axis to show complete graphs

karm-patel commented on notebooks/book1/03/gauss_infer_1d.ipynb:
```diff
@@ -0,0 +1,336 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #1.    latexify(width_scale_factor=2, fig_height=2.5)
</pre><p>please set fig_height = 2, I confirmed from latex source code</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/21/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

nitish1295 replied:
Done

karm-patel commented on notebooks/book1/03/gauss_infer_1d.ipynb:
```diff
@@ -0,0 +1,336 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #29.        plt.xticks([-7, 0, 7])
</pre><p>Try commenting this line, may be then x axis will have more ticks and it would be easy to interpret values in x-axis.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pyprobml-review/pyprobml-review/pull/21/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

nitish1295 replied:
<p>Done</p>

karm-patel commented:
Hi @nitish1295 , code looks good except few comments.

1. Can you fix this error? maybe keep lower values greater than zero.
![image](https://user-images.githubusercontent.com/59387624/165816733-27c397e5-2f25-4515-a5f0-d97b5bcdacad.png)

2. Plot is cutting for these values, we should fix ylim. 
![image](https://user-images.githubusercontent.com/59387624/165817164-67270f88-747b-43f9-b2fe-3e6e4b8a067a.png)

nitish1295 commented:
@karm-patel 

- [x] Fixed fig_height=2 and removed x-axis ticks
- [x] Fixed interactive plot axes limits to accommodate graphs
- [x] Fixed slider values for interactive plots

Updated figure
![image](https://user-images.githubusercontent.com/21181046/165829276-3695da28-0a87-4e33-9e2f-ab2538c810ea.png)
